src/main/java/com/snowflake/kafka/connector/records/RecordService.java

```diff
@@ -50,14 +50,20 @@
   private static final String KEY_SCHEMA_ID = "key_schema_id";
   static final String HEADERS = "headers";
 
-  public static final SimpleDateFormat ISO_DATE_FORMAT =
-      new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
-  public static final SimpleDateFormat TIME_FORMAT = new SimpleDateFormat("HH:mm:ss.SSSZ");
-  static final int MAX_SNOWFLAKE_NUMBER_PRECISION = 38;
+  // For each task, we require a separate instance of SimpleDataFormat, since they are not
+  // inherently thread safe
+  private static final ThreadLocal<SimpleDateFormat> ISO_DATE_FORMAT =
```

Comment:
Can we call this ISO_DATE_TIME_FORMAT as it also includes time information?

Reply:
Will do!